src/festim/exports/maximum_surface.py

```diff
@@ -27,11 +27,15 @@ def compute(self):
         subdomain, and appends it to the data list
         """
         solution = self.field.solution
-        indices = self.surface.locate_boundary_facet_indices(
-            solution.function_space.mesh
+        entities = self.facet_meshtags.find(self.surface.id)
+        V = solution.function_space
+        mesh = V.mesh
+        # mesh.topology.create_connectivity(mesh.topology.dim - 1, mesh.topology.dim - 1)
+        dofs = dolfinx.fem.locate_dofs_topological(
+            V=V, entity_dim=mesh.topology.dim - 1, entities=entities
         )
```

Comment:
This wont work in multispecies cases as the solution is an index in this case, not an fem.function. maybe entities could be an attribute of the SurfaceSubdomain or VolumeSubdomain?

Reply:
Hm I managed to make this work in a multispecies case

Reply:
Ah no sorry hang on

Reply:
So:

It works with multiple species using `HydrogenTransportProblem`

Derived quantities are not supported with `HydrogenTransportProblemDiscontinuous` yet anyway #980